test.js

```diff
@@ -114,6 +114,34 @@ test('test WriteStream with async writes', function (t, ctx, done) {
   write()
 })
 
+test('race condition between batch callback and close event', function (t, ctx, done) {
+  var down = ctx.db.db
+  var _batch = down._batch
+
+  // Delaying the batch should not be a problem
+  down._batch = function (ops, options, cb) {
+    setTimeout(function () {
```

Comment:
Note that `process.nextTick` seemed to be enough to trigger the race condition, the added delay is to be safe.

Reply:
Awesome! This is what I was thinking about today, to override the batching and delay it further! :heart: